packages/wdio-browserstack-service/src/service.js

```diff
@@ -0,0 +1,131 @@
+import logger from '@wdio/logger'
+import request from 'request'
+
+const log = logger('wdio-browserstack-service');
+
+export default class BrowserstackService {
+    constructor (config) {
+        this.config = config
+    }
+
+    before() {
+        this.sessionId = global.browser.sessionId;
+        this.failures = 0;
+        this.auth = global.browser.requestHandler ? global.browser.requestHandler.auth || {} : {};
+        return this._printSessionURL();
+    }
+
+    afterSuite(suite) {
+        if (suite.hasOwnProperty('error')) {
+            this.failures++;
+        }
+    }
+
+    afterTest(test) {
+        if (!test.passed) {
+            this.failures++;
+        }
+    }
+
+    afterStep(feature) {
+        if (
+            /**
+             * Cucumber v1
+             */
+            feature.failureException ||
+            /**
+             * Cucumber v2
+             */
+            (typeof feature.getFailureException === 'function' && feature.getFailureException()) ||
+            /**
+             * Cucumber v3, v4
+             */
+            (feature.status === 'failed')
+        ) {
+            ++this.failures
+        }
+    }
+
+    after() {
+        return this._update(this.sessionId, this._getBody());
+    }
+
+    onReload(oldSessionId, newSessionId) {
+        this.sessionId = newSessionId;
+        return this._update(oldSessionId, this._getBody())
+            .then(() => {
+                this.failures = 0;
+            })
+            .then(() => this._printSessionURL())
+    }
+
+    _update(sessionId, requestBody) {
+        return new Promise((resolve, reject) => {
+            request.put(`https://www.browserstack.com/automate/sessions/${sessionId}.json`, {
+                json: true,
+                auth: {
+                    user: this.auth.user,
+                    pass: this.auth.pass
+                },
+                body: requestBody
+            }, (error, response, body) => {
+                /* istanbul ignore if */
+                if (error) {
+                    return reject(error)
+                }
+                return resolve(body)
+            })
+        });
+    }
+
+    _getBody() {
+        return {
+            status: this.failures === 0 ? 'completed' : 'error'
+        };
+    }
+
+    _printSessionURL() {
+        const capabilities = global.browser.capabilities;
+        return new Promise((resolve,reject) => request.get(
+            `https://www.browserstack.com/automate/sessions/${this
+                .sessionId}.json`,
```

Comment:
why is here a new line?

Reply:
I think that was a direct copy and paste from the old repo (I think they used to use prettier with a line wrap rule)